packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -386,11 +397,11 @@ class IconButton extends StatelessWidget {
         hoverColor: hoverColor ?? theme.hoverColor,
         highlightColor: highlightColor ?? theme.highlightColor,
         splashColor: splashColor ?? theme.splashColor,
-        radius: splashRadius ?? math.max(
-          Material.defaultSplashRadius,
-          (iconSize + math.min(padding.horizontal, padding.vertical)) * 0.7,
-          // x 0.5 for diameter -> radius and + 40% overflow derived from other Material apps.
-        ),
+        radius: splashRadius ??
+            (autoImplySplashRadius
+                ? (iconSize + math.min(padding.horizontal, padding.vertical)) *
+                    0.7
+                : Theme.of(context).iconButtonSplashRadius),
```

Comment:
the previous logic here was to make sure we always splashed at least as big as the icon, if it was bigger than the default splash radius, or at least as big as the splash radius, if it was bigger than the icon, which I think is still the logic we want?

Basically it seems like the earlier algorithm is correct except that instead of "Material.defaultSplashRadius" it would use "Theme.of(context).iconButtonSplashRadius".

Reply:
Ohh, that seems right